app/src/org/commcare/android/javarosa/IntentCallout.java

```diff
@@ -303,10 +309,53 @@ private void storePointerToFileResponse(TreeReference ref, String responseValue,
         if (!src.exists()) {
             //TODO: How hard should we be failing here?
             Log.w(TAG, "CommCare received a link to a file at " + src.toString() + " to be included in the form, but it was not present on the phone!");
+
+            Uri uri = Uri.parse(responseValue);
+
+            try {
+                ParcelFileDescriptor inFile = androidContext.getContentResolver().openFileDescriptor(uri, "r");
+
+                File newFile = new File(destinationFile, uri.getLastPathSegment());
+
+                long fileLength = -1;
+
+                //Looks like our source file exists, so let's go grab it
+                try {
+                    FileChannel srcFc;
+                    srcFc = new FileInputStream(inFile.getFileDescriptor()).getChannel();
+                    fileLength = srcFc.size();
+                    FileChannel dst = new FileOutputStream(newFile).getChannel();
+                    dst.transferFrom(srcFc, 0, srcFc.size());
+                    srcFc.close();
+                    dst.close();
+                } catch (IOException e) {
+                    Log.e(TAG, "IOExeception copying Intent binary.");
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/dimagi/commcare-android/blob/c0d09222e8b6b707b1789a28cfddc833d6f2a4c3/app/src/org/commcare/android/javarosa/IntentCallout.java#L332"></span>Using 'Log' instead of 'Timber'</td>
    </tr>
  </tbody>
</table>